ls1intum/Artemis
# Add support for fetching entity titles to display as breadcrumbs

### Checklist

### Motivation and Context

For non-admins and non-instructors IDs would be displayed instead of titles in several cases, since normal users and teach assisstants didn't have the permissions to fetch the entity to read the title from.
Closes #2860.

### Description

We introduce new methods to fetch only the titles, such that no important data is leaked when an arbitrary user executes the query.

### Steps for Testing

Test that breadcrumbs display the correct names instead of IDs for:
- Courses
- Lectures
- Exams
- Exercises
- Exercise Hints
- Apollon Diagrams

For example, on develop tutors cannot fetch the entire exam, hence an error is displayed:
![grafik](https://user-images.githubusercontent.com/72132281/112169671-b29e5700-8bf2-11eb-8eb7-5780563ca236.png)
And the breadcrumbs look like this (notice the ID):
![grafik](https://user-images.githubusercontent.com/72132281/112169698-b8943800-8bf2-11eb-9b9d-67b81dfd6eda.png)

With this PR there is no error and the breadcrumbs look like the following:
![grafik](https://user-images.githubusercontent.com/72132281/112169759-c6e25400-8bf2-11eb-8a47-a533fa540106.png)

### Test Coverage

I adapted the client side navbar test which uses the new client side methods and introduced server-side tests for every new REST endpoint. => 100%
